acme/dns_challenge_rfc2136.go

```diff
@@ -2,31 +2,42 @@ package acme
 
 import (
 	"fmt"
-	"github.com/miekg/dns"
+	"strings"
 	"time"
+
+	"github.com/miekg/dns"
 )
 
 // DNSProviderRFC2136 is an implementation of the ChallengeProvider interface that
 // uses dynamic DNS updates (RFC 2136) to create TXT records on a nameserver.
 type DNSProviderRFC2136 struct {
-	nameserver string
-	zone       string
-	tsigKey    string
-	tsigSecret string
-	records    map[string]string
+	nameserver    string
+	zone          string
+	tsigAlgorithm string
+	tsigKey       string
+	tsigSecret    string
+	records       map[string]string
 }
 
 // NewDNSProviderRFC2136 returns a new DNSProviderRFC2136 instance.
-// To disable TSIG authentication 'tsigKey' and 'tsigSecret' must be set to the empty string.
-// 'nameserver' must be a network address in the the form "host:port". 'zone' must be the fully
+// To disable TSIG authentication 'tsigAlgorithm, 'tsigKey' and 'tsigSecret' must be set to the empty string.
+// 'nameserver' must be a network address in the the form "host" or "host:port". 'zone' must be the fully
 // qualified name of the zone.
-func NewDNSProviderRFC2136(nameserver, zone, tsigKey, tsigSecret string) (*DNSProviderRFC2136, error) {
+func NewDNSProviderRFC2136(nameserver, zone, tsigAlgorithm, tsigKey, tsigSecret string) (*DNSProviderRFC2136, error) {
+	// Append the default DNS port if none is specified.
+	if !strings.Contains(nameserver, ":") {
+		nameserver += ":53"
+	}
 	d := &DNSProviderRFC2136{
 		nameserver: nameserver,
 		zone:       zone,
 		records:    make(map[string]string),
 	}
-	if len(tsigKey) > 0 && len(tsigSecret) > 0 {
+	if len(tsigAlgorithm) > 0 && len(tsigKey) > 0 && len(tsigSecret) > 0 {
+		if tsigAlgorithm == "" {
```

Comment:
This case is always false as we do not enter this branch if `len(tsigAlgorithm)` is zero.